### PR TITLE
Present real failing checks on merge-queue kickout

### PR DESCRIPTION
**Pull request by subsetpark**
## Problem

When a patch's CI fails on its PR head, we hand the patch agent the **actual** failing check-runs (names, `details_url`, descriptions) via `render_ci_failure_prompt`. When a patch is **kicked out of the merge queue** ("github-merge-queue removed this pull request from the merge queue due to failed status checks"), we hand it only a synthetic `Ci_check.merge_queue_failure ()` placeholder — a generic description, no `details_url`, no dedup id. The agent never learns *which* checks failed.

**Root cause:** the poll query reads checks from `pullRequest.commits(last:1)` — the PR head. GitHub's merge queue runs CI on the ephemeral merge-group commit (the `gh-readonly-queue/...` head), a *different* SHA. For an ejected PR the PR-head checks are all green, so `apply_failure` falls back to the placeholder.

## Fix

This is deterministically fixable. `RemovedFromMergeQueueEvent.beforeCommit` is the merge-group commit GitHub actually ran checks on — the run shown in the PR timeline's "N of M checks passed" removal panel. (Confirmed against the live GraphQL schema via introspection: `beforeCommit : Commit`, `Commit.statusCheckRollup`, and the `REMOVED_FROM_MERGE_QUEUE_EVENT` timeline item type all exist.)

- New `Forge.merge_queue_removal_checks` fires a targeted GraphQL query for that commit's `statusCheckRollup`, reusing the existing rollup parser (`ci_check_of_context`) verbatim and returning only failures.
- At the existing re-sync seam in `runner_fiber_impl.ml` (which already isolates and preserves the synthetic check through the freshness gate), we prefer the real fetched checks over the placeholder, falling back on empty/error.
- Fires **only** when delivering a merge-queue CI failure (rare) — not on the hot poll path.

The pure decision layer (`merge_queue_decision.ml`, `types.ml`) is untouched: it still detects the ejection and injects the synthetic check as the durable signal/fallback; the IO layer upgrades the content when it can. Real checks carry real `databaseId`s, so CI-feedback dedup now works for merge-queue failures (the placeholder had `id = None`, which always re-delivered).

## Result

The kickout prompt becomes symmetric with the normal CI path — real check names, `Details` links, and dedup ids.

## Testing

- New `test/test_github_merge_queue_removal.ml`: mixed pass/fail rollup (only failures returned, with real ids/urls), empty timeline, null `beforeCommit`, missing PR, and GraphQL-error cases.
- `dune build` clean; full `dune test` suite green (pre-commit hook enforces build + test + format).
- Schema verified via `gh api graphql` introspection.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Improved PR status reporting with support for merge-queue removal events, surfacing the most relevant failing checks when available.

* **Bug Fixes**
  * Failing checks now prefer real merge-queue results over placeholder values when possible, making CI feedback more accurate.
  * Added safer handling for missing or incomplete merge-queue data so PR checks still display cleanly.

* **Tests**
  * Added coverage for merge-queue check parsing and edge cases.

### CODE REVIEW

File: lib/runner_fiber_impl.ml [hunk 1]
```diff
@@ -1398,9 +1398,42 @@ struct
...
+                                              with
+                                              | Ok (_ :: _ as real) ->
+                                                  log_event runtime ~patch_id
+                                                    (Printf.sprintf
+                                                       "Fetched %d failing \
+                                                        merge-queue check(s) \
+                                                        from removal event"
+                                                       (Base.List.length real));
+                                                  real
+                                              | Ok [] | Error _ ->
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply lib/runner_fiber_impl.ml]: Fixed by splitting the merge-queue removal check fallback: `Ok []` now logs the empty-result placeholder path, while `Error e` logs `Forge.show_error e` before falling back to the synthetic placeholder.

File: lib/github.ml [hunk 2]
```diff
@@ -675,6 +675,63 @@ let parse_enqueue_pr_info_response body =
...
+type merge_queue_removal_pull_request = {
+  timeline_items : removal_timeline_items;
+      [@key "timelineItems"] [@yojson.default { nodes = [] }]
+}
+[@@deriving of_yojson] [@@yojson.allow_extra_fields]
+
+let checks_of_commit (c : commit) : Types.Ci_check.t list =
+  match c.status_check_rollup with
+  | None -> []
+  | Some rollup -> List.filter_map rollup.contexts.nodes ~f:ci_check_of_context
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply lib/github.ml]: Fixed. `checks_of_commit` now treats `statusCheckRollup.contexts.pageInfo.hasNextPage = true` as non-authoritative and returns `[]`, so the caller keeps the synthetic merge-queue placeholder instead of replacing it with a partial first page. I also added a regression fixture covering a truncated removal-event rollup.

File: lib/runner_fiber_impl.ml [hunk 3]
```diff
@@ -1398,9 +1398,42 @@ struct
...
+                                                    "No merge-queue removal \
+                                                     checks available — using \
+                                                     placeholder";
+                                                  synthetic_checks)
+                                          | None -> synthetic_checks
+                                      in
                                       let ci_checks =
                                         pr_state.Pr_state.ci_checks
-                                        @ synthetic_checks
+                                        @ merge_queue_checks
```
[the remark this answers is not in the document]
Comment (subsetpark) [reply lib/runner_fiber_impl.ml:1452]: Fixed. When removal-event checks are fetched successfully, the runner now stores the existing synthetic merge-queue marker alongside the real failures (`synthetic_checks @ real`) so downstream freshness/retry logic and `patch_controller` still see `Ci_check.is_merge_queue_failure` while the agent also gets the real check details.
